tests/unit/frameworks/vernon/self_supervised_learning_test.py

```diff
@@ -0,0 +1,124 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2021, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+
+import unittest
+
+import torch
+import torch.nn as nn
+import torchvision.transforms as transforms
+from torchvision.datasets.fakedata import FakeData
+
+from nupic.research.frameworks.pytorch.self_supervised_utils import EncoderClassifier
+from nupic.research.frameworks.vernon import SelfSupervisedExperiment
+
+
+class AutoEncoder(torch.nn.Module):
+    def __init__(self, input_dim=784, hidden_dim=20):
+        super().__init__()
+        self.fc1 = nn.Linear(input_dim, hidden_dim)
+        self.fc2 = nn.Linear(hidden_dim, input_dim)
+
+    def forward(self, x):
+        encoded = self.encode(x)
+        decoded = self.fc2(encoded).view(-1, 1, 28, 28)
+        return decoded
+
+    def encode(self, x):
+        x = x.flatten(start_dim=1)
+        encoded = self.fc1(x)
+        return encoded
+
+
+class LinearClassifier(torch.nn.Module):
+    def __init__(self, input_dim=20, num_classes=10):
+        super().__init__()
+        self.fc = nn.Linear(input_dim, num_classes)
+
+    def forward(self, x):
+        out = self.fc(x)
+        return out
+
+
+# # MNIST
```

Comment:
These comments can be removed, I presume.

Reply:
Yes, removed.